### ISSUE TITLE
functions.php

### ISSUE THREAD

**Issue by evolutic** [evolutic/wp-themes#1]
<?php
//http://wpchannel.com/supprimer-blog-permaliens-wordpress-3-0-multi-site
add_filter("pre_update_option_category_base","cc_remove_blog_slug");
add_filter("pre_update_option_tag_base","cc_remove_blog_slug");
add_filter("pre_update_option_permalink_structure","cc_remove_blog_slug");
function cc_remove_blog_slug($tag_cat_permalink){
	if(!preg_match("/^\/blog\//",$tag_cat_permalink))
		return $tag_cat_permalink;
	$new_permalink=preg_replace ("/^\/blog\//","/",$tag_cat_permalink );
	return $new_permalink;
}
?>
<?php
require_once('assets/cpt.php'); 
//require_once('wp-advanced-search/wpas.php');
?>
<?php
/**
 * Marta Y Paula functions and definitions.
 *
 * Sets up the theme and provides some helper functions, which are used in the
 * theme as custom template tags. Others are attached to action and filter
 * hooks in WordPress to change core functionality.
 *
 * When using a child theme (see http://codex.wordpress.org/Theme_Development
 * and http://codex.wordpress.org/Child_Themes), you can override certain
 * functions (those wrapped in a function_exists() call) by defining them first
 * in your child theme's functions.php file. The child theme's functions.php
 * file is included before the parent theme's file, so the child theme
 * functions would be used.
 *
 * Functions that are not pluggable (not wrapped in function_exists()) are
 * instead attached to a filter or action hook.
 *
 * For more information on hooks, actions, and filters,
 * see http://codex.wordpress.org/Plugin_API
 *
 * @package WordPress
 * @subpackage Marta y Paula
 * @since Marta y Paula 1.0
 */
?>
<?php
// Este tema usa wp_nav_menu() en una localización.
	register_nav_menu( 'principal', __( 'men&uacute; de navegaci&oacute;n', 'martaypaula' ) );
	register_nav_menu( 'secundario', __( 'men&uacute; secundario de navegaci&oacute;n', 'martaypaula' ) );

/* Google Analytics */
add_action('wp_head', 'add_google_analytics');
function add_google_analytics() {
?>
<meta name="google-site-verification" content="I-iiw3leXkEhcTrKc5XNnCr8hwJmdOnPPnDfACurzHo" />
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-47048185-1', 'martaypaula.es');
  ga('require', 'linkid', 'linkid.js');
  ga('send', 'pageview');
</script>
<?php
}

/* Borra los párrafos vacíos de las entradas del content y del excerpt */
  remove_filter( 'the_content', 'wpautop' ); //para el contenido
  remove_filter( 'the_excerpt', 'wpautop' ); //para el extracto

/* Elimina de wp_head */
  remove_action('wp_head', 'rsd_link');
  remove_action('wp_head', 'wp_generator');
  remove_action('wp_head', 'feed_links', 2);
  remove_action('wp_head', 'index_rel_link');
  remove_action('wp_head', 'wlwmanifest_link');
  remove_action('wp_head', 'feed_links_extra', 3);
  remove_action('wp_head', 'start_post_rel_link', 10, 0);
  remove_action('wp_head', 'parent_post_rel_link', 10, 0);
  remove_action('wp_head', 'adjacent_posts_rel_link', 10, 0);
  remove_action('wp_head', 'wp_shortlink_wp_head', 10, 0);

/* using Contact Form 7 3.9 or higher, you can also disable the loading of the JavaScript and CSS by adding a few lines of code into your theme’s functions.php */
  add_filter( 'wpcf7_load_js', '__return_false' );
  add_filter( 'wpcf7_load_css', '__return_false' ); 

/**
 * Cargar Scripts.
 *
 * @since Yoedu 3.0
 * 
 * @return void
 *
 */
  add_action('wp_head', 'carga_js_WPCF7', 1);
  function carga_js_WPCF7() {
    if ( is_page_template('formularios.php') ) {
        //echo "\r\n";
       if ( function_exists('wpcf7_enqueue_scripts') ) {
          //echo "\r\n";
          /** Contact Form 7 Removing BR tags */
          define("WPCF7_AUTOP",false);
          wpcf7_enqueue_scripts();
      }
    }
  }

/**
 * Registra dos áreas para widgets.
 *
 * @since Marta y Paula 1.0
 *
 * @return void
 *
 */

//add_action( 'widgets_init', 'martaypaula_widgets_init' );	

function martaypaula_widgets_init() {

	register_sidebar( array(
		'name' => __( 'Principal', 'martaypaula' ),
		'id' => 'sidebar-1',
		'before_widget' => '<aside id="%1$s" class="widget %2$s">',
		'after_widget' => "</aside>",
		'before_title' => '<h3 class="widget-title">',
		'after_title' => '</h3>',
	) );

	register_sidebar( array(
		'name' => __( 'Secundario', 'martaypaula' ),
		'id' => 'sidebar-2',
		'before_widget' => '<aside id="%1$s" class="widget %2$s">',
		'after_widget' => "</aside>",
		'before_title' => '<h3 class="widget-title">',
		'after_title' => '</h3>',
	) );
}

/**
 * Elimina el estilo css de recent comments
 *
 */

//add_action( 'widgets_init', 'martaypaula_remove_recent_comments_style' );
function martaypaula_remove_recent_comments_style() {
	global $wp_widget_factory;
	remove_action( 'wp_head', array( $wp_widget_factory->widgets['WP_Widget_Recent_Comments'], 'recent_comments_style' ) );
}
?>
<?php
/**
 * Reemplzar el formulario por defecto de búsqueda por una versión HTML5
 *
 */
//add_filter( 'get_search_form', 'martaypaula_html5_search_form' );
function martaypaula_html5_search_form( $form ) {
    $form = '<form role="search" method="get" id="searchform" action="' . home_url( '/' ) . '" >
    <label class="assistive-text" for="s">' . _x('b&uacute;squeda provincia o c&oacute;digo postal', 'martaypaula') . '</label>
    <input required type="search" placeholder="'. _x('provincia o c&oacute;digo postal', 'martaypaula').'" value="' . gen_slug(get_search_query()) . '" name="s" id="s" />
    <button type="submit">Buscar</button>
    </form>';

    return $form;
}

//add_action('wp_head', 'debug_scripts_queued');
function debug_scripts_queued() {
    global $wp_scripts;
    echo '";
}

/**
 * Cargar Scripts.
 *
 * @since Yoedu 3.0
 * 
 * @return void
 *
 */

  add_action('wp_enqueue_scripts', 'carga_scripts');
  function carga_scripts() {

    if( ! is_admin() ) {

        wp_deregister_script('jquery');
        // handle, registrar la ubicación, dependencias y versión de su script, infooter.
        wp_register_script('jquery', 'http://ajax.googleapis.com/ajax/libs/jquery/1/jquery.min.js', false, false, false);

        /*version que deregistra estilos master-slider*/
        add_filter( 'masterslider_enqueue_styles', '__return_false' );
        
        /*ley-cookie (no está cumpliendo con la citada ley)
        wp_dequeue_style('ley-cookie');
        wp_deregister_style('ley-cookie');
        */

        /*version que deregistra vegas*/
        //wp_dequeue_script('vegas');
        //wp_dequeue_style('vegas');
        //wp_deregister_script('vegas');
        //wp_deregister_style('vegas');        
        //wp_register_script('vegas', 'http://jaysalvat.github.io/vegas/releases/latest/jquery.vegas.min.js', array('jquery'), null, false);
        //wp_register_script('vegas', plugins_url('../js/jquery.vegas.js',__FILE__), array('jquery'), null, 1 );
        // handle, registrar la ubicación, dependencias y versión de su script, infooter.
        //wp_register_style('vegas', '//jaysalvat.github.io/vegas/releases/latest/jquery.vegas.min.css', '', null, 'all'); 
        //remove_shortcode( 'vegas', array( $this, 'vegas' ) );      

        wp_register_script('modernizr', get_bloginfo('template_directory') . '/js/modernizr.js', false, false, false );
        //wp_register_script('hammer', get_bloginfo('template_directory') . '/js/hammer.min.js', false, false, true );
        //wp_register_script('scripthammer', get_bloginfo('template_directory') . '/js/scripthammer.js', array('hammer'), false, true );
        //wp_register_script('scriptresponsivemenu', get_bloginfo('template_directory') . '/js/scriptresponsivemenu.js', array('jquery'), false, true );
        //wp_register_script('nicescroll', get_bloginfo('template_directory') . '/js/jquery.nicescroll.min.js', array('jquery'), false, true );        
        //wp_register_script('scriptnicescroll', get_bloginfo('template_directory') . '/js/scriptnicescroll.js', array('nicescroll'), false, true );          
        wp_enqueue_script('modernizr');
        // pone en cola al script
        wp_enqueue_script('jquery');
        //add_shortcode( 'vegas', array( $this, 'vegas' ) );    

        //wp_enqueue_script('scriptnicescroll');  
        //wp_enqueue_script('nicescroll'); 
        //wp_enqueue_script('hammer');
        //wp_enqueue_script('scripthammer');
        //wp_enqueue_script('scriptresponsivemenu');
          

        if( is_singular('colecciones') ) { 
          echo "\n";
          //add_filter( 'masterslider_enqueue_styles'. plugin_basename(__FILE__), 'my_custom_masterslider_enqueue_styles' );
          //add_filter( 'plugin_action_links_' . plugin_basename(__FILE__), 'my_plugin_action_links' );
          add_filter( 'masterslider_enqueue_styles', 'my_custom_masterslider_enqueue_styles', 10, 1 );
       
            function my_custom_masterslider_enqueue_styles( $styles_queue ){
                $styles_queue['masterslider-main'] = array(
                    'src'     => plugins_url('/master-slider/public/assets/css/masterslider.main.css'),
                    'deps'    => array(),
                    'version' => '1.0'
                );
                // make sure to return $styles_queue at the end
                return $styles_queue;
            }
        }   
        if( is_page() ) { 
          echo "\n";
        }     
        if( is_home() ) { 
          echo "\n";
        }
        if( is_front_page() ) { 
          echo "\n";
        }
        if( is_front_page() && is_home() || is_page_template('template-supersized.php') || is_page() || is_category() || is_single() || is_page_template('formularios.php') ) {
          echo "\n";
          // handle, registrar la ubicación, dependencias y versión de su script, infooter.
          wp_deregister_script('jquery');
          wp_register_script('jquery', 'https://ajax.googleapis.com/ajax/libs/jquery/1/jquery.min.js', false, true, false);
          wp_register_script('jquery-easing', get_bloginfo('template_directory') . '/slideshow/js/jquery.easing.min.js', array('jquery'), true, false);
          wp_register_script('supersized', get_bloginfo('template_directory') . '/slideshow/js/supersized.3.2.7.min.js', array('jquery', 'jquery-easing'), true, false);
          
          wp_enqueue_script('jquery');
          wp_enqueue_script('jquery-easing');
          wp_enqueue_script('supersized');
        }
        if( is_tax('temporadas') ) {
          echo "\n";
          // handle, registrar la ubicación, dependencias y versión de su script, infooter.
          wp_deregister_script('jquery');
          wp_deregister_script('jquery', 'https://ajax.googleapis.com/ajax/libs/jquery/1/jquery.min.js', false, true, false);
          wp_deregister_script('jquery-easing', get_bloginfo('template_directory') . '/slideshow/js/jquery.easing.min.js', array('jquery'), true, false);
          wp_deregister_script('supersized', get_bloginfo('template_directory') . '/slideshow/js/supersized.3.2.7.min.js', array('jquery', 'jquery-easing'), true, false);
        }
        if( is_singular('colecciones') ) {
          echo "\n";
          // handle, registrar la ubicación, dependencias y versión de su script, infooter.
          wp_deregister_script('supersized', get_bloginfo('template_directory') . '/slideshow/js/supersized.3.2.7.min.js', array('jquery', 'jquery-easing'), true, false);
        }        
        //sigue la cola aquí   

    }
  }

?>
